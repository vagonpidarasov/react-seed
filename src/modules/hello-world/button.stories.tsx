import React from 'react';
import {HelloWorldComponent} from './button';

export default {component: HelloWorldComponent, title: 'Button'};
export const defaultButton = () => <HelloWorldComponent/>;
