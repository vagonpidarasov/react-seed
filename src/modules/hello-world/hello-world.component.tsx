import React from 'react';
import styled from '@emotion/styled';

type HelloWorldPropsType = {content:string};
const Container = styled.div`
    padding: 1em;
`;

export const HelloWorldComponent:React.FC<HelloWorldPropsType> = ({content}) =>
    <Container className="hello-world" data-test-id="hello-world">{content}</Container>;
