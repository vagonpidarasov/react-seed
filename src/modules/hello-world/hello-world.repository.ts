import {Observable} from 'rxjs';

export abstract class HelloWorldRepository {
    abstract getContent():Observable<string>;
}
