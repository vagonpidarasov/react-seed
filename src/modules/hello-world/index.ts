export * from './hello-world.container';
export * from './hello-world.repository';
export * from './redux';
