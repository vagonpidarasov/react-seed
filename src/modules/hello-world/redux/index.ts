export * from './epics';
export * from './reducer';
export * from './state';
export * from './feature';
export * from './selectors';
export * from './actions';
