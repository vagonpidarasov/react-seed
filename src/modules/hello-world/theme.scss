$color: 000 !default;
$background-color: transparent !default;

.hello-world {
    color: $color;
    background-color: $background-color;
}
