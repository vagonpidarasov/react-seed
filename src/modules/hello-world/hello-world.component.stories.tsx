import React from 'react';
import {HelloWorldComponent} from './hello-world.component';

export default {component: HelloWorldComponent, title: 'Button'};
export const defaultButton = () => <HelloWorldComponent content="Hello World"/>;
