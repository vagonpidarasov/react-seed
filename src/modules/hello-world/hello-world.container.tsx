import React from 'react';
import {useSelector} from 'react-redux'
import {HelloWorldComponent} from './hello-world.component';
import {content as selectContent} from './redux/selectors';

export const HelloWorldContainer:React.FC = () => {
    const content = useSelector(selectContent);
    return <HelloWorldComponent content={content}/>;
};
