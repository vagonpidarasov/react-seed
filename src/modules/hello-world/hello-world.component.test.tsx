import React from 'react';
import {render} from '@testing-library/react';
import {HelloWorldComponent} from './hello-world.component';

describe('HelloWorldComponent', () => {
    test('matches snapshot', async () => {
        const {container} = render(<HelloWorldComponent content="hello world"/>);
        expect(container).toMatchSnapshot();
    });

    test('renders content', async () => {
        const {container} = render(<HelloWorldComponent content="hello world"/>);
        expect(container.firstChild.textContent).toBe('hello world');
    });
});
