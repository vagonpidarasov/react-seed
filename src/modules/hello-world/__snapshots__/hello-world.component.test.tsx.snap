// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldComponent matches snapshot 1`] = `
.emotion-0 {
  padding: 1em;
}

<div>
  <div
    class="hello-world emotion-0"
    data-test-id="hello-world"
  >
    hello world
  </div>
</div>
`;
